File: agc/agc016/agc016_a/agc016_a.py
import string

s = input()
flag = True
for c in s:
  if s[0] != c:
    flag = False

if flag:
  print(0)
else:
  ans = 1e9
  ptn = list(string.ascii_lowercase)
  
  for c in ptn:
    cnt = 0
    x = s

    while len(x) > 1:
      cnt += 1
      y = ""

      for i in range(len(x) - 1):
        if x[i] == c or x[i + 1] == c:
          y += c
        else:
          y += x[i]
      
      flag = True
      for a in y:
        if y[0] != a:
          flag = False
      
      if flag:
        break
      x = y

    ans = min(ans, cnt)
  
  print(ans)

File: agc/agc016/agc016_a/agc016_a.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  cin.tie(0);
  ios_base::sync_with_stdio(0);

  string s;
  cin >> s;

  set<char> se;
  for(auto &z : s) se.insert(z);
  if(se.size() == 1) {
    cout << 0;
    return 0;
  }

  int ans = 1e9;
  for(char c = 'a'; c <= 'z'; c++) {
    int cnt = 0;
    string x = s;

    while(x.size() > 1) {
      cnt++;
      string y = "";

      for(int i = 0; i < x.size() - 1; i++) {
        if(x[i] == c || x[i + 1] == c) y += c;
        else y += x[i];
      }

      set<char> se;
      for(auto &z : y) se.insert(z);
      if(se.size() == 1) break;

      x = y;
    }

    ans = min(ans, cnt);
  }

  cout << ans;
  return 0;
}